library/src/main/api21/com/google/android/cameraview/Camera2.java

```diff
@@ -435,11 +450,49 @@ protected void collectPictureSizes(SizeMap sizes, StreamConfigurationMap map) {
 
     private void prepareImageReader() {
         Size largest = mPictureSizes.sizes(mAspectRatio).last();
-        mImageReader = ImageReader.newInstance(largest.getWidth(), largest.getHeight(),
+        initImageReader(largest.getWidth(), largest.getHeight());
+
+        Size size = chooseOptimalSize();
+        initCallbackImageDataReader(size.getWidth(), size.getHeight());
+    }
+
+    private void initImageReader(int width, int height){
+        mImageReader = ImageReader.newInstance(width, height,
                 ImageFormat.JPEG, /* maxImages */ 2);
         mImageReader.setOnImageAvailableListener(mOnImageAvailableListener, null);
     }
 
+    private void initCallbackImageDataReader(int width, int height) {
+
+        if (mCallbackImageDataReader != null) {
+            mCallbackImageDataReader.close();
+        }
+
+        mCallbackImageDataReader = ImageReader.newInstance(width, height,
+                ImageFormat.NV21, /* maxImages */ 1);
```

Comment:
Hi, I don't think `NV21` is supported in `Camera2`. It should be `YUV_420_888` in `Camera2` case. It cashed on my N5X with `Caused by: java.lang.IllegalArgumentException: NV21 format is not supported`